### PR TITLE
RATIS-1692. Remove the use of the thirdparty Counter.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1692

### CODE REVIEW

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java [hunk 1]
```diff
@@ -33,16 +32,14 @@
...
-  Counter counter(String name);
+  LongCounter counter(String name);
 
   boolean remove(String name);
 
   <T> void gauge(String name, Supplier<Supplier<T>> gaugeSupplier);
 
   Timer timer(String name, MetricRegistry.MetricSupplier<Timer> supplier);
 
-  Counter counter(String name, MetricRegistry.MetricSupplier<Counter> supplier);
```
Comment (codings-dan) [inline ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java:44]: While I don't find any calls in ratis code, I'm not sure if removing it directly would introduce compatibility issues.

**Review by codings-dan — COMMENTED** [review]
@szetszwo Thanks for working on this, overall lgtm, just left one comment for reference, thanks!

File: ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java (hunk 1 above)
Comment (szetszwo) [reply ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java:44]: @codings-dan , thanks for reviewing this.

- First of all, ratis-metrics is not yet a public API.  It is neither in the ratis-server-api module nor in the org.apache.ratis.protocol package.
- The JIRA is to remove the use Counter.  There is no way to make it API compatible (e.g. Changing Counter to LongCounter is already API incompatible).  The user applications (e.g. Ozone) using it must update their code.

We may consider NOT merging this to 2.4.0 for now.  Once we have finished all the sub tasks in  RATIS-1688, we may merge all the changes to branch-2 and require the user applications to update their code.

Comment (codings-dan) [reply ratis-metrics/src/main/java/org/apache/ratis/metrics/RatisMetricRegistry.java:44]: > * There is no way to make it API compatible
Got it. 

> We may consider NOT merging this to 2.4.0 for now
Sure, I also think 2.4.0 doesn't need to include RATIS-1688 related code.

**Review by codings-dan — APPROVED** [review]
The change looks good, +1

**Comment by szetszwo** [comment]
@codings-dan , thanks a lot for merging this!

**Pull request merged**